### ISSUE TITLE
INFO -  modifications Rumba32 board

### ISSUE THREAD

**Issue by DrumClock** [makerbase-mks/MKS-RUMBA32#34]
Hi, below are possible modifications :

1) **use jumpers for DIAG** settings instead of R82-R87
   possibility to connect jumper to Endstop or wire as needed e.g.

     XDIAG = X- (jumper)
     YDIAG = Y- (jumper)
     ZDIAG = Z + (wire for Z-max position)
     E2DIAG = Y + (wire, for IDEX mode)

2) **remove R90-R95** for UART-mod use:

 ```
#define X_SERIAL_TX_PIN PC14
 #define X_SERIAL_RX_PIN PC14

  #define Y_SERIAL_TX_PIN PE4
  #define Y_SERIAL_RX_PIN PE4
  
  #define Z_SERIAL_TX_PIN PE0
  #define Z_SERIAL_RX_PIN PE0

  #define E0_SERIAL_TX_PIN PC13
  #define E0_SERIAL_RX_PIN PC13
 
 #define E1_SERIAL_TX_PIN PD5
  #define E1_SERIAL_RX_PIN PD5
 
  #define E2_SERIAL_TX_PIN PD1
  #define E2_SERIAL_RX_PIN PD1
```

3) **create new EXP 4** connector
    from point   TP3: PB10，TP4: PA15，TP5: PD11
  
 ```
**
 *             ---                                           
 *       PD11 | · |              
 *        GND | · |---             
 * (PWM) PA15 | · | · | PB10 (PWM)            
 *         5V | · | · | 5V              
 *        GND | · | · | GND                 
 *             -------                                          
 *              EXP4

```

4)  **change the pin for BTN_EN2** to other than PB2 (BOOT1) !!
      info: #33 (comment)

###    **I will test everything and give information**

**Comment by DrumClock** [comment]
### Hi, my tests:

**point 1)** - StallGuard cannot be set for Z axis in Marlin FW - others work.

**point 2)** - UART mode for drivers works well and pins on conector EXP3 can be used (eg, for BL-touch, servo, etc.)

**point 3)**  create EXP4 from points TP3, TP4, TP5

![BL NE](https://user-images.githubusercontent.com/64655979/107958100-d3471180-6fa1-11eb-9547-42a22e2d9baf.jpg)


**point 4)**  remapping BTN pin from PB2 to PB3

For BigTreeTech TFT Touch Screen with using EXP1 and EXP2 connection.
 Otherwise it is not possible to start DFU mod for uploading FW Marlin ! 

![PB3](https://user-images.githubusercontent.com/64655979/107958121-d80bc580-6fa1-11eb-94a8-5cafb1c91609.jpg)

**Comment by mks-viva** [comment]
Thank you, we will consider it.

**Comment by DrumClock** [comment]
OK and change the pin for BTN to other than PB2 (BOOT1) !!

**Comment by DrumClock** [comment]
The Rumba board (mega2560 or STM32) is, in my opinion, the most equipped for various variants of 3D printers.

So far I have not found any that would allow 6 motors, 5 temperature sensors, and 5 HE / FAN switches .....

I would just make a few HW changes and it will be the best board ....

**Comment by DrumClock** [comment]
### Here is my design layout on the board 

![PIN out spolecne](https://user-images.githubusercontent.com/64655979/108032367-a7c03780-7032-11eb-885d-1abb84057d3a.jpg)

**Comment by FinnsMax** [comment]
Hey, what are the R90-R95 resistors on the board for? I want to use tmc2209 uart, but I do not plan such a deep redesign of the board. Can resistors immediately delete and reassign the pins?

**Comment by DrumClock** [comment]
Hey @FinnsMax 
If you do not want to use the EXP3 connector with the TMC2209-UART mode as well, you do not need to anything.

information: https://github.com/makerbase-mks/MKS-RUMBA32/issues/31#issuecomment-768449817

**Comment by FinnsMax** [comment]
Ок

**Comment by Cobactan57** [comment]
Hi, in the current setting could you point a pin I could use as data pin for neopixel ? Thanks !

**Comment by DrumClock** [comment]
> Hi, in the current setting could you point a pin I could use as data pin for neopixel ? Thanks !

Hi, the description is how I created a new EXP4 connector for BL-touhc and NeoPixel.
It's tried and tested.

**Comment by Cobactan57** [comment]
Thanks, but sorry it's not clear to me: can I use PB10 as neopixel pin without modifying to get an Exp4 ? If so where is PB10 on the board ? Is it TP3 ?  How about using PD14 on Exp3 ? Many thanks !

**Comment by DrumClock** [comment]
> Thanks, but sorry it's not clear to me: can I use PB10 as neopixel pin without modifying to get an Exp4 ? If so where is PB10 on the board ? Is it TP3 ? How about using PD14 on Exp3 ? Many thanks !


1)  TP3: PB10，TP4: PA15，TP5: PD11，you can define it to other function

2)  PD12,PD13,PD14, PD15, on EPX3 you can also use but only if you do not have TMC2209 UART mode.

**Comment by Cobactan57** [comment]
Will try, Thanks a lot !

**Comment by DrumClock** [comment]
I have tried it ..... it works :)

**Comment by FinnsMax** [comment]
![IMG_20210309_191510](https://user-images.githubusercontent.com/78873432/110491654-4e1fca00-8113-11eb-9717-5f761476b29e.jpg)
![IMG_20210309_191533](https://user-images.githubusercontent.com/78873432/110491680-5415ab00-8113-11eb-82b4-3e10ecfbcc97.jpg)
![IMG_20210309_192013](https://user-images.githubusercontent.com/78873432/110491690-56780500-8113-11eb-9604-fea65e3e4790.jpg)

**Comment by FinnsMax** [comment]
Servo0 --> TP4 --> NC AUX1
100 K -->B2 EXP2 (GND) --> https://github.com/Aus3D/RUMBA32/blob/master/Resources/Images/RUMBA32_v1_1B_Schematic.png

**Comment by DrumClock** [comment]
> 100 K -->B2 EXP2 (GND) -->

Hi @FinnsMax 
I tried it but if the TFT was connected via EXP1 and EXP2 board could not 
switch to DFU mode to load Marlin, so I remapped the PB2 PIN.

Is this how it works for you?

**Comment by FinnsMax** [comment]
With TFT has not yet tried. Works with LCD RepRap

**Comment by DrumClock** [comment]
Hi, I know that from LCD12864 it's no problem :)
The problem occurs with the FTF display there is always level H on pin PB2.

That's why I had to modify the MKS Rumba32 board.

**Comment by FinnsMax** [comment]
I have Hi only of BTT TFT35 V3.0, will try soon

**Comment by DrumClock** [comment]
> Thank you, we will consider it.

Hi,
will you be releasing a new version of Rumba32 soon?
Or should I buy version 1.0 and remake it again?

**Comment by T0bbi3** [comment]
Ive been all in for buying rumba32 as i got other mks products. But for now I'll wait some time for a new revision of the board then buy other brand if nothing is released. Doesn't seem to be any huge faults but I'm no good in fine electronics/soldering. People will pay more for a fixed board. Please give some release info on this🙈

Connectors should also be added for:
1x Mks pwc module
1x Mks ups module
1x Neopixel

Thanks in advance

**Comment by DrumClock** [comment]
Hi @mks-viva 
BIGTREETECH has released a new OCTOPUS-V1.0 board.
Are you also planning to use a board with these parameters?
If so, BTT lacks a connector for connecting two other servos (except BLTouch)
